src/components/PrimaryNav/PrimaryNav.js

```diff
@@ -1,72 +1,78 @@
-import React from "react";
+import React, { useState } from "react";
 import { Link } from "react-router-dom";
+import classNames from "classnames";
 
-function PrimaryNav() {
+import "./_primary-nav.scss";
+
+const pages = [
+  { label: "Models", path: "/" },
+  { label: "Controllers", path: "/controllers" },
+  { label: "Usage", path: "/usage" },
+  { label: "Logs", path: "/logs" }
+];
+
+const PrimaryNav = () => {
+  const currentLocation = window.location.pathname;
+  const [extNavOpen, setExtNavOpen] = useState(false);
   return (
-    <div>
-      <header id="navigation" className="p-navigation">
-        <div className="p-navigation__banner">
-          <div className="p-navigation__logo">
-            <a className="p-navigation__link" href="/">
-              <img
-                className="p-navigation__image"
-                src="https://assets.ubuntu.com/v1/a9e0ed4a-jaas-logo1.svg"
-                alt="JAAS logo"
-                width="95"
-              />
-            </a>
-          </div>
+    <nav
+      className={classNames("p-primary-nav", { "ext-nav-open": extNavOpen })}
+    >
+      <div className="p-primary-nav__header">
+        <Link to="/">
+          <img
+            className="p-primary-nav__logo"
+            src="https://assets.ubuntu.com/v1/a9e0ed4a-jaas-logo1.svg"
+            alt="JAAS logo"
+          />
+        </Link>
+        <button
+          className="p-primary-nav__toggle"
+          onClick={() => setExtNavOpen(!extNavOpen)}
+        >
+          <i className="p-icon--contextual-menu"></i>
```

Comment:
You can add text inside an icon for accessibility which Vanilla will hide.

Reply:
That's not mentioned in the docs - added an issue: https://github.com/canonical-web-and-design/vanilla-framework/issues/2573